core/src/main/java/org/apache/brooklyn/entity/group/SensorMatchingRemovalStrategy.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.brooklyn.entity.group;
+
+import java.util.Collection;
+import java.util.Objects;
+
+import javax.annotation.Nullable;
+
+import org.apache.brooklyn.api.entity.Entity;
+import org.apache.brooklyn.api.sensor.AttributeSensor;
+import org.apache.brooklyn.config.ConfigKey;
+import org.apache.brooklyn.core.config.ConfigKeys;
+
+import com.google.common.reflect.TypeToken;
+
+public class SensorMatchingRemovalStrategy<T> extends RemovalStrategy {
+    public static final ConfigKey<AttributeSensor> SENSOR = ConfigKeys.newConfigKey(new TypeToken<AttributeSensor>() {}, "sensor.matching.sensor");
+    // Would be nice to use ConfigKey<T>, but TypeToken<T> cannot be instantiated at runtime
+    public static final ConfigKey<Object> DESIRED_VALUE = ConfigKeys.newConfigKey(new TypeToken<Object>() {}, "sensor.matching.value");
```

Comment:
@neykov scarily, changing the use of TypeToken will renumber all the anonymous inner classes below it! If anyone has persisted those, it will cause problems.

We really do need to get to be stricter by default, that we won't accept persisting anonymous inner classes!